portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,22 +25,36 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    
+  static final ArrayList<String> comments = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    ArrayList<String> hardcode = new ArrayList<>();
-    hardcode.add("fire");
-    hardcode.add("water");
-    hardcode.add("earth");
-
     response.setContentType("text/html;");
-    response.getWriter().println(convertToJsonUsingGson(hardcode));
+    response.getWriter().println(convertToJsonUsingGson(comments));
   }
-  private String convertToJsonUsingGson(ArrayList<String> hardcode){
+  private String convertToJsonUsingGson(ArrayList<String> comments){
     Gson gson = new Gson();
```

Comment:
Since the Gson object can be reused rather than recreated every time this function is called, you can make it a class member (similar to what you've done with the list of comments) and save time by not creating it on every Get request.

https://sites.google.com/site/gson/gson-user-guide#TOC-Using-Gson